Source/KISAddonConfig.cs

```diff
@@ -22,52 +23,133 @@ sealed class KISAddonConfig : MonoBehaviour {
   const string MaleKerbalEva = "kerbalEVA";
   const string FemaleKerbalEva = "kerbalEVAfemale";
   const string RdKerbalEva = "kerbalEVA_RD";
-  
-  public void Awake() {
-    ConfigAccessor.ReadFieldsInType(GetType(), this);
-    ConfigAccessor.ReadFieldsInType(typeof(ModuleKISInventory), instance: null);
-
-    // Set inventory module for every eva kerbal
-    DebugEx.Info("Set KIS config...");
-    ConfigNode nodeSettings = GameDatabase.Instance.GetConfigNode("KIS/settings/KISConfig");
-    if (nodeSettings == null) {
-      DebugEx.Error("KIS settings.cfg not found or invalid !");
-      return;
+
+  static ConfigNode nodeSettings;
+
+  class KISConfigLoader: LoadingSystem
+  {
+    public KISAddonConfig owner;
+
+    public override bool IsReady ()
+    {
+      return true;
+    }
+
+    public override void StartLoad ()
+    {
+      ConfigAccessor.ReadFieldsInType(owner.GetType(), owner);
+      ConfigAccessor.ReadFieldsInType(typeof(ModuleKISInventory), instance: null);
+
+      // Set inventory module for every eva kerbal
+      DebugEx.Info("Set KIS config...");
+      nodeSettings = GameDatabase.Instance.GetConfigNode("KIS/settings/KISConfig");
+      if (nodeSettings == null) {
+        DebugEx.Error("KIS settings.cfg not found or invalid !");
+      }
     }
+  }
+
+  class KISPodInventoryLoader: LoadingSystem
+  {
+    public bool done;
+
+    int loadedPartCount;
+    int loadedPartIndex;
+
+    IEnumerator LoadInventories ()
+    {
+      // Kerbal parts.
+      UpdateEvaPrefab(MaleKerbalEva, nodeSettings);
+      UpdateEvaPrefab(FemaleKerbalEva, nodeSettings);
 
-    // Kerbal parts.
-    UpdateEvaPrefab(MaleKerbalEva, nodeSettings);
-    UpdateEvaPrefab(FemaleKerbalEva, nodeSettings);
-
-    // Set inventory module for every pod with crew capacity.
-    DebugEx.Info("Loading pod inventories...");
-    foreach (AvailablePart avPart in PartLoader.LoadedPartsList) {
-      if (avPart.name == MaleKerbalEva || avPart.name == FemaleKerbalEva
-          || avPart.name == RdKerbalEva
-          || !avPart.partPrefab || avPart.partPrefab.CrewCapacity < 1) {
-        continue;
+      // Set inventory module for every pod with crew capacity.
+      DebugEx.Info("Loading pod inventories...");
+      for (loadedPartIndex = 0; loadedPartIndex < loadedPartCount; loadedPartIndex++) {
+        AvailablePart avPart = PartLoader.LoadedPartsList[loadedPartIndex];
+        if (!(avPart.name == MaleKerbalEva || avPart.name == FemaleKerbalEva
+              || avPart.name == RdKerbalEva
+              || !avPart.partPrefab || avPart.partPrefab.CrewCapacity < 1)) {
+          DebugEx.Fine("Found part with CrewCapacity: {0}", avPart.name);
+          AddPodInventories (avPart.partPrefab, avPart.partPrefab.CrewCapacity);
+        }
+        yield return null;
```

Comment:
Returning `null` will result in loading one part per a frame. On the low-end machines with many addons it may take forever to complete. This is definitely not the case when smooth loading is better than quick loading.  At least do some batching here. And, in general, let's extract this refactoring into a separate PR. I'd like to test it before saying "yes".